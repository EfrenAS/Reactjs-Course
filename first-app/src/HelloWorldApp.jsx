import SayHello from "./SayHello.jsx";

export default function HelloWorldApp() {
  return (
    <>
      <SayHello 
        name="Efrén"
        lastname="Anastacio"
        city="Santiago de Querétaro"
      />
    </>
  );
}
