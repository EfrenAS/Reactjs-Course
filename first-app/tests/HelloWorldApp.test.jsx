import { render, screen } from "@testing-library/react"
import HelloWorldApp from "../src/HelloWorldApp.jsx"

describe("Pruebas en <HelloWorldApp />", () => {
  const title = "Hello Efrén Anastacio"
  
  test("Debe de hacer match con el snapshot", () => {
    const { container } = render(<HelloWorldApp />)
    
    expect(container).toMatchSnapshot()
  })

  test("Debe de mostrar el nombre de Efrén dentro de un h1", () => {
    render(<HelloWorldApp />)
    // const h1 = container.querySelector("h1")

    // expect(h1.innerHTML).toBe(title)
    
    expect(screen.getByText(title)).toBeTruthy()
    expect(screen.getByTestId("test-title").innerHTML).toContain(title)
  })
})