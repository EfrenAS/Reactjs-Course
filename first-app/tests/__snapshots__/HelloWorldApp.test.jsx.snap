// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <HelloWorldApp /> Debe de hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hello 
    Efrén
     
    Anastacio
  </h1>
  <p>
    You are from 
    Santiago de Querétaro
  </p>
</div>
`;
